### ISSUE TITLE
Webpack build: Critical dependency: require function is used in a way in which dependencies cannot be statically extracted

### ISSUE THREAD

**Issue by dakur** [scriptex/touchsweep#127]
I'd say there must be something broken with the built file in `dist/` folder in v2. With v1.3 it worked well.

I use ES modules everywhere and typescript (even though it failed in .js file as well).

Output from webpack build with `--stats-error-details`:

```
WARNING in ./node_modules/touchsweep/dist/touchsweep.js 3:24-31
Critical dependency: require function is used in a way in which dependencies cannot be statically extracted
    at CommonJsRequireContextDependency.getWarnings ([myproject]\node_modules\webpack\lib\dependencies\ContextDependency.js:102:18)
    at Compilation.reportDependencyErrorsAndWarnings ([myproject]\node_modules\webpack\lib\Compilation.js:3132:24)
    at [myproject]\node_modules\webpack\lib\Compilation.js:2729:28
    at eval (eval at create ([myproject]\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:29:1)
    at [myproject]\node_modules\webpack\lib\FlagDependencyExportsPlugin.js:385:11
    at [myproject]\node_modules\neo-async\async.js:2830:7
    at Object.each ([myproject]\node_modules\neo-async\async.js:2850:39)
    at [myproject]\node_modules\webpack\lib\FlagDependencyExportsPlugin.js:361:18
    at [myproject]\node_modules\neo-async\async.js:2830:7
    at Object.each ([myproject]\node_modules\neo-async\async.js:2850:39)
    at [myproject]\node_modules\webpack\lib\FlagDependencyExportsPlugin.js:51:16
    at Hook.eval [as callAsync] (eval at create ([myproject]\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:25:1)
    at Hook.CALL_ASYNC_DELEGATE [as _callAsync] ([myproject]\node_modules\tapable\lib\Hook.js:18:14)
    at Compilation.finish ([myproject]\node_modules\webpack\lib\Compilation.js:2714:28)
    at [myproject]\node_modules\webpack\lib\Compiler.js:1182:19
    at processTicksAndRejections (node:internal/process/task_queues:78:11)
 @ [somepath]/FlashMessage.ts 2:19-40
 @ [somepath]/index.js 8:0-67 53:2-14

```

**Comment by dakur** [comment]
On reddit, there is an explanation to it: https://www.reddit.com/r/typescript/comments/4q09uv/help_webpack_weirdness_require_function_is_used/

I think it might be the reason. According to another comment, it could be enough changing `module: "umd"` to `module: "commonjs"` in `tsconfig.json`, but I'm not quite familiar with js packages export settings, so I don't know. 🤷‍♂️

**Comment by scriptex** [comment]
Hi, @dakur, is it possible to paste your webpack config or at least provide a minimal reproducible example?

**Comment by dakur** [comment]
Hi @scriptex, thank you for your answer. I'm able to reproduce it with this minimal config:

```js
module.exports = {
	entry: './index.js',
}
```

I've also removed `.babelrc` to be sure that no other lib beside webpack is in effect.

**Comment by scriptex** [comment]
🤔 I was able to reproduce this and to be honest, I am not sure why this happens. 
It looks like Webpack doesn't like UMD produced by the TypeScript compiler...
Can you try adding the following to your Webpack config:
```
module: {
  unknownContextCritical: false,
},
```
[Webpack's docs](https://webpack.js.org/configuration/module/#module-contexts) do not say much about this configuration setting, though.

**Comment by dakur** [comment]
Yes, `unknownContextCritical: false` does the job, but.. 🙂

The aforementioned reddit thread wasn't helpful? It seems to me that this could be the problem:

> You need to tell Typescript to emit CommonJS, not UMD. UMD is doing funky stuff with the require function, passing it as an argument into factory. This means webpack can't look at all usages of the top-level require function and figure out exactly what modules you're loading.

It well corresponds to generated touchsweep code in `dist/` folder which appeared in my `node_modules/` after installing v2:

<img src="https://user-images.githubusercontent.com/9862581/199059367-50f1f63b-93fa-4c71-afbb-b8cdc3370d10.png" width=600>

**Comment by scriptex** [comment]
Okay, looks like I need to change the way the module is built. I guess I will try Vite or Rollup but I don’t have the time right now. 
How critical is your situation? 😅

**Comment by dakur** [comment]
Not at all, I'll wait. Thank you very much!

**Comment by scriptex** [comment]
@dakur I've released a new version (2.0.1) on NPM and on [Github](https://github.com/scriptex/touchsweep/releases/tag/2.0.1). 
The new release uses Rollup to build and (according to my tests) fixes the Webpack warning.
Please check and let me know 🙂
Thank you for your contribution!

**Comment by dakur** [comment]
Yes, that works very well now, thank you very much!
